### PR TITLE
Added pre-commit hook for Lua scripts from Policy folder

### PR DESCRIPTION
**Pull request by aderiabin**
https://github.com/CustomSDL/sdl_atf_test_scripts/pull/34/

### CODE REVIEW

**Comment by aderiabin** [comment]
@LuxoftAKutsan please review my pull request

File: tools/lua_style_and_check.sh [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+git diff --check --cached --color -- .$GIT_DIFF_CHECK_LIST | cat
+if [ "$?" -ne "0" ]; then
+  echo -e "$TEXT_ERROR" "Your changes introduce whitespace errors"
+  echo -e " Aborting commit." "$TEXT_DEFAULT"
+
+  exit 1
+fi
+echo -e "$TEXT_INFO" "PASSED" "$TEXT_DEFAULT"
+
+##################################################################
```
Comment (LuxoftAKutsan) [inline tools/lua_style_and_check.sh]: @aderiabin please avoid such borders around the comments

File: tools/lua_style_and_check.sh [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+echo -e "$TEXT_INFO" "Checking lua code with luacheck" "$TEXT_DEFAULT"
+
+LUA_CHECK=$(command -v luacheck) 
+
+if [ ! -x "$LUA_CHECK" ]; then
+  echo -e "$TEXT_ERROR" "Error: luacheck executable not found." "$TEXT_DEFAULT"
+  echo -e " Aborting commit." "$TEXT_DEFAULT"
+  exit 2
```
Comment (LuxoftAKutsan) [inline tools/lua_style_and_check.sh]: @aderiabin Is you use some return code, please specify them in the beginning of the script with variables

File: .luacheckrc [hunk 3]
```diff
@@ -0,0 +1,15 @@
+globals = {"Test",
+  "config",
```
Comment (aderiabin) [inline .luacheckrc]: @LuxoftAKutsan 
Common way to use "config" variable is use a global variable
Example:  
```
config.deviceMAC = "<MAC_value>"
```

File: tools/git-hooks/pre-commit [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    trap POP_STASHED EXIT
+fi
+# End of stashing
+
+if git rev-parse --verify HEAD >/dev/null 2>&1
+then
+    against=HEAD
+else
+    # Initial commit: diff against an empty tree object
+    against=4b825dc642cb6eb9a060e54bf8d69288fbee4904
```
Comment (aderiabin) [inline tools/git-hooks/pre-commit]: @LuxoftAKutsan Replaced on initial commit in c7bb3d62f7daa8fbe83ad799c9467954cd7e0912

File: tools/lua_style_and_check.sh [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+
+##################################################################
+### Check for odd whitespace
+##################################################################
+
+echo -e "$TEXT_INFO" "Checking odd whitespaces" "$TEXT_DEFAULT"
+git diff --check --cached --color -- .$GIT_DIFF_CHECK_LIST | cat
+if [ "$?" -ne "0" ]; then
+  echo -e "$TEXT_ERROR" "Your changes introduce whitespace errors"
```
Comment (aderiabin) [inline tools/lua_style_and_check.sh]: @LuxoftAKutsan quotes were removed in c7bb3d62f7daa8fbe83ad799c9467954cd7e0912

File: tools/lua_style_and_check.sh (hunk 1 above)
Comment (aderiabin) [inline tools/lua_style_and_check.sh]: @LuxoftAKutsan comments were simplified in c7bb3d62f7daa8fbe83ad799c9467954cd7e0912

File: tools/lua_style_and_check.sh (hunk 2 above)
Comment (aderiabin) [inline tools/lua_style_and_check.sh]: @LuxoftAKutsan All return codes were specified with variables in c7bb3d62f7daa8fbe83ad799c9467954cd7e0912

**Pull request merged**